bus-mapping/src/circuit_input_builder/execution.rs

```diff
@@ -336,6 +336,39 @@ pub enum NumberOrHash {
     Hash(H256),
 }
 
+/// Represents all bytes related in one copy event.
+#[derive(Clone, Debug)]
+pub struct CopyBytes {
```

Comment:
Really three buffers are needed:
- Reader data.
- Writer data.
- Writer data before the write.

And two masks:
- Reader mask.
- Writer mask.

It’s not clear to me which is which in this structure.

Reply:
i realize  `bytes_read_prev` is not necessary. removed in [b2f568d](https://github.com/scroll-tech/zkevm-circuits/pull/569/commits/b2f568d33fed5deb389f3d3cebc6e8f14132a98b)

Reply:
So:

```
bytes                   = reader data
aux_bytes            = writer data after write
bytes_write_prev = writer data before write
```

Then `aux_bytes` could be named to `bytes_write`

Do you interpret these fields differently depending on whether it is memory-to-memory or not?